File: Python/DataStructure/grapgh/stack/stack.py
class stack(object):
    def __init__(self): 
        self.items = []

    def isEmpty(self):
        return len(self.items) == 0

    def length(self):
        return len(self.items)
    
    def push(self, element):
        self.items.append(element)

    def top(self, element, underflow):
        if self.isEmpty():
            underflow = True
        else:
            underflow = False
            element = self.items[len(self.items) - 1]
        return element, underflow

    def pop(self, element, underflow):
        if isEmpty():
            underflow = True
        else:
            underflow = False
            element = self.items.pop()
        return element, underflow

    def display(self):
        for i in range(len(self.items)):
            print(self.items[i])

File: Java/Data Structures/Stack/Stack.java
class Stack<T>{
    class Node<T>{
        T data;
        Node next;

        public Node(T data){
            this.data = data;
        }
    }

    Node head;

    public boolean isEmpty(){
        return head == null;
    }
    
    public void push(T data){
        Node node = new Node(data);
        node.next = head;
        head = node;
    }

    public T pop(){
        Node temp;
        if(isEmpty()) 
            return null;
        temp = head;
        head = head.next;
        return (T) temp.data;
    }

    public T top() throws Exception{
        if(isEmpty()){
            throw new java.lang.RuntimeException("Queue is Empty");
        }
        return (T) head.data;
    }
}